phpcs.xml.dist

```diff
@@ -23,89 +23,41 @@ limitations under the License.
+	<rule ref="WordPress-Docs">
 		<exclude-pattern>tests/*</exclude-pattern>
 	</rule>
-	<rule ref="Squiz.Commenting.FunctionCommentThrowTag.Missing">
-		<exclude-pattern>tests/*</exclude-pattern>
-	</rule>
-
-	<!-- Do not apply filename rules for unit tests -->
-	<rule ref="WordPress.Files.FileName.NotHyphenatedLowercase">
-		<exclude-pattern>tests/*</exclude-pattern>
-	</rule>
-	<rule ref="WordPress.Files.FileName.InvalidClassFileName">
+	<rule ref="WordPress-Extra">
+		<exclude name="Generic.CodeAnalysis.UselessOverridingMethod"/>
+		<exclude name="WordPress.Files.FileName"/>
+		<exclude name="Universal.CodeAnalysis.ConstructorDestructorReturn.ReturnTypeFound"/>
+		<exclude name="Universal.CodeAnalysis.ConstructorDestructorReturn.ReturnValueFound"/>
+		<exclude name="Universal.NamingConventions.NoReservedKeywordParameterNames.defaultFound"/>
 		<exclude-pattern>tests/*</exclude-pattern>
 	</rule>
+	<rule ref="WordPress.WP.I18n"/>
+	<config name="text_domain" value="wp-third-parties,default"/>
 
-	<!-- Do not apply compatibility rules to allow using the modern PHPUnit functionality -->
-	<rule ref="PHPCompatibility.FunctionDeclarations.NewReturnTypeDeclarations.stringFound">
-		<exclude-pattern>tests/utils/*</exclude-pattern>
-	</rule>
-	<rule ref="PHPCompatibility.FunctionDeclarations.NewReturnTypeDeclarations.boolFound">
-		<exclude-pattern>tests/utils/*</exclude-pattern>
-	</rule>
-
-	<!-- Disallows grouped use declarations. -->
 	<rule ref="SlevomatCodingStandard.Namespaces.DisallowGroupUse" />
-	<!-- Disallows leading backslash in use statement. -->
 	<rule ref="SlevomatCodingStandard.Namespaces.UseDoesNotStartWithBackslash" />
-	<!-- Checks whether uses at the top of a file are alphabetically sorted. -->
 	<rule ref="SlevomatCodingStandard.Namespaces.AlphabeticallySortedUses" />
-
-	<!-- Prohibits uses from the same namespace. -->
 	<rule ref="SlevomatCodingStandard.Namespaces.UseFromSameNamespace" />
-	<!-- Looks for unused imports from other namespaces. -->
 	<rule ref="SlevomatCodingStandard.Namespaces.UnusedUses">
-	<properties>
-	  <property name="searchAnnotations" value="true" />
-	</properties>
+		<properties>
+			<property name="searchAnnotations" value="true" />
+		</properties>
 	</rule>
-	<!-- All references to functions, classes and constants should import using a use statement. -->
-	  <rule ref="SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly">
-		 <properties>
-			<property name="allowFullyQualifiedGlobalFunctions" value="true" />
-			<property name="allowFullyQualifiedGlobalClasses" value="true" />
-			<property name="allowFullyQualifiedGlobalConstants" value="true" />
-			<property name="allowFallbackGlobalFunctions" value="true" />
-			<property name="allowFallbackGlobalConstants" value="true" />
-			<property
-			name="allowFullyQualifiedNameForCollidingClasses"
-			value="true"
-		  />
+
+	<rule ref="Generic.Files.LineLength">
+		<properties>
+			<property name="lineLimit" value="120"/>
```

Comment:
curious where 120 is from, does core use that value?

Reply:
Core doesn't enforce this anywhere, and I'm not even sure it's written anywhere, but it's extremely common to see lines limited at 120 characters. This is mostly notable in function docblocks with longer descriptions or parameter descriptions.

I think this probably happens because it's a common convention (e.g. in https://www.php-fig.org/psr/psr-2/#:~:text=There%20MUST%20NOT%20be%20a%20hard%20limit%20on%20line%20length%3B%20the%20soft%20limit%20MUST%20be%20120%20characters%3B%20lines%20SHOULD%20be%2080%20characters%20or%20less.), so either many developers know about it or just follow what they assume to be the line length limit from other core code.